Tools/Scripts/webkitpy/style/checkers/basexcconfig.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2013-2022 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+"""Enforces rules for Base.xcconfig files."""
+
+
+class BaseXcconfigChecker(object):
```

Comment:
I think this rule would be even better if it dynamically parsed setting names from `CommonBase.xcconfig`. Otherwise, I expect people will move things into CommonBase without knowing to add a clause to the rule.

At initialization, we could do something like:
```python
self.common_settings_and_extensions = {}
for line in open('Configurations/CommonBase.xcconfig'):
    # Find lines containing assignments.
    lhs, operator, rhs = map(str.strip, line.partition('='))
    # Skip non-assignment lines, comments, and WK_ variables which are allowed to be overridden.
    if operator != '=' or lhs.startswith('//') or lhs.startswith('WK_'):
        continue
    # Discard any setting condition.
    name = lhs.partition('[')[0]
    suggested_extension_name = f'WK_EXTRA_{name}' if 'WK_EXTRA_' in rhs else f'WK_DEFAULT_{name}'
    self.common_settings_and_extensions[name] = suggested_extension_name
```

Reply:
Great idea!

With what we learned about how `$(inherited)` works with XCBuild, I think we can tweak the code above to put non-`WK_` variables that include `$(inherited)` in a different category, and then verify that `$(inherited)` also appears in the same variable(s) in `Base.xcconfig`.

Reply:
This works so well!